### ISSUE TITLE
Commit composer.lock

### ISSUE THREAD

**Issue by mirfilip** [thephpleague/flysystem-sftp#44]
Hi,

I just noticed this repo doesn't have a lock file commited and, as of writing this, specifies a loose version constraint for its most important dependency: [phpseclib/phpseclib](https://github.com/phpseclib/phpseclib):
```
"phpseclib/phpseclib": "~2.0"
```

It means, when a user includes a particular **tagged** version of `flysystem-sftp`, `phpseclib/phpseclib` version is going to vary with the time, depending when versions are resolved.

It makes debugging development vs production stuff confusing.

Is that an oversight or am I missing the greater scheme?

**Comment by frankdejonge** [comment]
Hi @mirfilip,

We set the constraint like this so we're not tied to phpseclib's release schedule and users can "freely" get updates from that package without us holding it back. Apart from that lock files in this repo won't be used when the package is installed. Lock files are only used by composer for the current project, so for users that would be their own project. Lock files of their dependencies are ignored.

For your project, the one you deploy, you __should always__ commit your lock file. This will also ensure the phpseclib version deployed to your production environment.

I hope this cleared it up for you.

**Comment by mirfilip** [comment]
Hi @frankdejonge, thx for a quick answer.

I realised I simplified my exact use case and now it looks I don't know how to go about lock file :)

Of course, I always commit lock files for my projects , [unless I write a library](https://getcomposer.org/doc/02-libraries.md#lock-file).

My actual use case I came across this problem: I wanted a playground to test flysystem with sftp, so I set up a dummy "test" composer project and included both tagged version of `flysystem` and `flysystem-sftp`. At that point I presumed including tagged versions will lock dependencies to particular versions (that's obviously not true unless I force particular versions myself; just forgot about that fact). That way, when I updated packages a week later, I got a different version of `phpseclib/phpseclib`. My mistake was not including a lock file in my playground project.

Giving users an ability for update without you bumping version is an interesting approach from library owner's POV. I didn't think about that. If that works for you, great.

All is clear now.
